test/index.js

```diff
@@ -78,6 +78,11 @@ def ('$26')
     (a => b => c => d => e => f => g => h => i => j => k => l => m => n => o => p => q => r => s => t => u => v => w => x => y => z =>
        [a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t, u, v, w, x, y, z]);
 
+const reset = ((checkTypes, ...types) => () => {
+  $.checkTypes = checkTypes;
+  $.env.splice (0, Infinity, ...types);
+}) ($.checkTypes, ...$.env);
```

Comment:
Ideally this function would be called in a `finally` clause like so:

```javascript
try {
  $.env.push (Foo, Bar);
  eq (...) (...);
  eq (...) (...);
  eq (...) (...);
} finally {
  reset ();
}
```

This would ensure that `$.checkTypes` and `$.env` are restored to their default states even if an assertion fails, avoiding the possibility of future `test` block failing erroneously.

I opted not to do this, however, as it would add noise to the diff, increase indentation, and introduce some large `try` clauses. In this case I think the “incorrect” option is simpler and clearer. Do you agree?